d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancerState.java

```diff
@@ -582,6 +591,12 @@ public void done(int type, String name)
     return _clusterInfo;
   }
 
+  Map<String, LoadBalancerStateItem<ClusterFailoutProperties>> getClusterFailoutProperties()
```

Comment:
Why is the map being exposed here? I would think it would be better to expose a method to manipulate the map so that this class can make guarantees around mutability and thread safety. Is this just following the pattern for ServiceProperties?

Reply:
Yeah, was following `getClusterInfo`. Fixed.